lib/hammer_cli/ca_cert_fetcher.rb

```diff
@@ -1,30 +1,68 @@
+require 'fileutils'
+
 module HammerCLI
   class CACertFetcher
-    def self.fetch_ca_cert(host)
-      CACertFetcher.new.fetch_ca_cert(host)
+    attr_reader :hostname, :uri, :cert_file, :port, :host, :ca_path
+    def self.fetch_ca_cert(host, ca_path)
+      CACertFetcher.new(host, ca_path).fetch_ca_cert
+    end
+
+    def initialize(host, ca_path)
+      @host = host
+      @ca_path = ca_path
+      @uri = URI.parse(host)
+
+      @hostname = uri.host
+      @port = uri.port || 443
+      @cert_file = File.join(ca_path, "#{hostname}.pem")
+
+      if hostname.nil?
+        $stderr.puts _("Couldn't parse URI '%s'.") % host
+        $stderr.puts scheme_error(uri)
+        return HammerCLI::EX_SOFTWARE
+      end
     end
 
-    def fetch_ca_cert(host)
+    def fetch_ca_cert
       begin
-        uri = URI.parse(host)
+        raw_cert = get_ca_cert
+        store_ca_cert(raw_cert)
+
+        rh_install_path = "/etc/pki/ca-trust/source/anchors/"
+        rh_update_cmd = "update-ca-trust"
+        deb_install_path = "/usr/local/share/ca-certificates/"
+        deb_update_cmd = "update-ca-certificates"
+
+        puts _("CA certificate for #{hostname} was stored to #{cert_file}")
+        puts _("Now hammer can use the downloaded certificate to verify SSL connection to the server.")
+        puts
+        puts _("Be aware that hammer cannot verify whether the certificate is correct and you should verify its authenticity.")
+        puts
+        puts _("You can display the certificate content with")
+        puts "  $ openssl x509 -text -in #{cert_file}"
+        puts
 
-        hostname = uri.host
-        port = uri.port || 443
+        cert_install_msg = "\n" + _("As root you can also install the certificate and update the system-wide list of trusted CA certificates as follows:") + "\n"
 
-        if hostname.nil?
-          $stderr.puts _("Couldn't parse URI '%s'.") % host
-          $stderr.puts scheme_error(uri)
-          return HammerCLI::EX_SOFTWARE
+        if File.directory?(rh_install_path)
+          puts cert_install_msg
+          puts "  $ install #{cert_file} #{rh_install_path}"
+          puts "  $ #{rh_update_cmd}"
         end
+        if File.directory?(deb_install_path)
```

Comment:
I agree, but I don't understand how the pattern `IF rh_path_exist THEN print_rh_instructions ELSIF deb_path exist THEN print_deb_instructions` breaks it.